table/tables/partition_test.go

```diff
@@ -700,6 +702,1460 @@ func TestIssue31629(t *testing.T) {
+				},
+				{
+					"SELECT count(*) FROM tkey_string WHERE id7 > 'large' AND id7 < 'x-small'",
+					false, false, true, true, []string{"partition:p2", "partition:p0", "partition:p3"}, []string{"partition:p1"}, 3,
+				},
+			},
+		},
+		{
+			partitionbySQL: "PARTITION BY KEY(id8) partitions 4",
+			selectInfo: []compoundSQL{
+				{
+					"SELECT count(*) FROM tkey_string",
+					false, false, true, true, []string{"partition:p0", "partition:p1", "partition:p2", "partition:p3"}, []string{}, 5,
+				},
+				{
+					"SELECT count(*) FROM tkey_string PARTITION(p0)",
+					false, false, true, true, []string{"partition:p0"}, []string{"partition:p1", "partition:p2", "partition:p3"}, 2,
+				},
+				{
+					"SELECT count(*) FROM tkey_string PARTITION(p1)",
+					false, false, true, true, []string{"partition:p1"}, []string{"partition:p0", "partition:p2", "partition:p3"}, 1,
+				},
+				{
+					"SELECT count(*) FROM tkey_string PARTITION(p2)",
+					false, false, true, true, []string{"partition:p2"}, []string{"partition:p1", "partition:p0", "partition:p3"}, 1,
+				},
+				{
+					"SELECT count(*) FROM tkey_string PARTITION(p3)",
+					false, false, true, true, []string{"partition:p3"}, []string{"partition:p1", "partition:p2", "partition:p0"}, 1,
+				},
+				{
+					"SELECT count(*) FROM tkey_string WHERE id8 = 'a'",
+					false, false, true, true, []string{"partition:p0"}, []string{"partition:p1", "partition:p2", "partition:p3"}, 2,
+				},
+				{
+					"SELECT count(*) FROM tkey_string WHERE id8 = 'a' or id8 = 'b'",
+					false, false, true, true, []string{"partition:p0", "partition:p3"}, []string{"partition:p1", "partition:p2"}, 3,
+				},
+				{
+					"SELECT count(*) FROM tkey_string WHERE id8 > 'a' AND id8 < 'c'",
+					false, false, true, true, []string{"partition:p1", "partition:p2", "partition:p0", "partition:p3"}, []string{}, 1,
+				},
+			},
+		},
+	}
+	executePartTableCase(t, tk, testCases3, createSQL3, insertSQLS3, dropSQL3)
+}
+
+func TestKeyPartitionTableMixed(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("create database partitiondb2")
+	defer tk.MustExec("drop database partitiondb2")
+	tk.MustExec("use partitiondb2")
+	// SHOW CREATE TABLE
+	tk.MustExec("CREATE TABLE tkey1 (col1 INT NOT NULL, col2 DATE NOT NULL,col3 INT NOT NULL, col4 INT NOT NULL, UNIQUE KEY (col3))" +
+		" PARTITION BY KEY(col3)" +
+		"(PARTITION `p0`," +
+		"PARTITION `p1`," +
+		"PARTITION `p2`," +
+		"PARTITION `p3`)")
+	tk.MustQuery("show create table tkey1").Check(testkit.Rows("tkey1 CREATE TABLE `tkey1` (\n" +
+		"  `col1` int(11) NOT NULL,\n" +
+		"  `col2` date NOT NULL,\n" +
+		"  `col3` int(11) NOT NULL,\n" +
+		"  `col4` int(11) NOT NULL,\n" +
+		"  UNIQUE KEY `col3` (`col3`)\n" +
+		") ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin\n" +
+		"PARTITION BY KEY (`col3`) PARTITIONS 4"))
```

Comment:
If the partitions have different names or have placement rules, would the output be the same?

Reply:
It's different. The testcase has already beeb added in this test file.